### PR TITLE
Update to new simd-json API for reused allocations

### PR DESCRIPTION
**Pull request by Licenser**
# Pull request

Update json codec to take advantage of @CJP10's improved deserialiser API.

## Description

The updated API does avoid two allocations per decoding cycle as it keeps buffers as part of the codec. This leads to a ~3% performance gain on the `realworld-througput-json` benchmark.

before: Throughput: `426.2 MB/s`
after: Throughput: `439.0 MB/s`

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 #![forbid(warnings)]
 #![deny(missing_docs)]
-#![recursion_limit = "1024"]
+#![recursion_limit = "2048"]
```
Comment (mfelsche) [inline src/lib.rs:19]: lol, feeling the pain that got you here

**Review by darach — APPROVED** [review]
LGTM 👍

**Review by darach — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (Licenser) [reply src/lib.rs:19]: ja and good catch lets see if that can be reverted now

Comment (Licenser) [reply src/lib.rs:19]: yes it can! Thanks for catching this!

**Review by mfelsche — APPROVED** [review]

**Pull request merged**